src/ffi/logging.rs

```diff
@@ -25,6 +26,44 @@ pub const START_ASYNC_LOGGER_MSG: &str = "Starting async logger thread";
 
 static DROPPED_LOGS: AtomicUsize = AtomicUsize::new(0);
 static DROPPED_LOGS_LAST_CHECKED_VALUE: AtomicUsize = AtomicUsize::new(0);
+pub static LOGGER_STOPPER: LoggerStopper = LoggerStopper::new();
+
+pub struct LoggerStopper {
+    sender: parking_lot::Mutex<Option<SyncSender<LogMessage>>>,
+}
+
+impl LoggerStopper {
+    const fn new() -> Self {
+        Self {
+            sender: parking_lot::Mutex::new(None),
+        }
+    }
+
+    fn set_sender(&self, sender: SyncSender<LogMessage>) {
+        *self.sender.lock() = Some(sender);
+    }
+
+    pub fn stop(&self) {
+        match *self.sender.lock() {
+            Some(ref sender) => {
+                let start = Instant::now();
+                let b = Arc::new(Barrier::new(2));
+                match sender.try_send(LogMessage::Quit(b.clone())) {
+                    Ok(()) => {
+                        b.wait();
```

Comment:
I'm thinking could two threads calling this cause a deadlock, this place feel's safeish, as sender is exclusive to one thread, and you wait for a barier. but if somehow we get a both FfiCallback drop and stop called at the same time from differnet threads, on barier will never be awaited for no ?

Reply:
On the other hand you are either using drop, or you are using with_default, but not both at the same time

Reply:
Well for test's we could be using stop + drop + with_default, but it's a more controlled enviroment

Reply:
Making queue size of 1 would ensure safety, but we would drop to many messages :/

Reply:
Oh maybe not as queue is per sender, if i remember correctly

Reply:
hmm, i guess it's finish. Although i feel it would be safer to simply have a lock on barrier it self, basicly trying to ensure that only one thread can wait hold it at a time. But it's strart feeling like over compication for non reachable code :sweat_smile:, leaving to you to decide. and resolve this.